adoc/chapters/architecture.adoc

```diff
@@ -1204,8 +1209,10 @@ work-group in <<local memory,local>> or <<global memory>> and to synchronize bet
 work-items in the same work-group by calling the
 [code]#group_barrier# function. All work-groups in a given
 [code]#parallel_for# will be the same size, and the global size
-defined in the nd-range must be a multiple of the work-group size in
-each dimension.
+defined in the nd-range must either be a multiple of the work-group size in
+each dimension, or the global size must be zero. In the case the global size
```

Comment:
```suggestion
each dimension, or the global size must be zero. When the global size
```

I think this is more concise.  If you prefer your original wording, I think there was some word missing.  Maybe you meant "In the case **where** the global size ...".